File: src/atr_pkg/src/ma_node.cpp
#include <ros/ros.h>
#include <iostream>
#include <std_msgs/String.h>

void chao_callback(std_msgs::String msg)
{
    std::cout << msg.data << std::endl;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "ma_node");
    std::cout << "马可上线了" << std::endl;
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("lets_go_to_play", 10, chao_callback);

    while (ros::ok())
    {
        ros::spinOnce();//用于不断检查之前的回调函数能不能继续调用
    }

    return 0;
}

File: src/atr_pkg/scripts/ma_node.py
#! /usr/bin/env python3
#coding=utf-8
import rospy
from std_msgs.msg import String

def chao_callback(msg: String):
    rospy.loginfo(msg.data)

def yao_callback(msg: String):
    rospy.logwarn(msg.data)


if __name__=="__main__":
    rospy.init_node("ma_node")

    sub=rospy.Subscriber("lets_go_to_play",String,chao_callback,queue_size=10)
    sub2=rospy.Subscriber("help_me_get_more_score",String,yao_callback,queue_size=10)

    rospy.spin()#这个保持节点在运行状态
